internal/config/configuration.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+var configurationVersion = "0.1"
+
+type Configuration struct {
+	Version  string `yaml:"version"`
+	Currency string `yaml:"currency,omitempty"`
+}
+
+func loadConfiguration(cfg *Config) error {
+	var err error
+
+	err = cfg.migrateConfiguration()
+	if err != nil {
+		logrus.Debug("Error migrating configuration")
+		logrus.Debug(err)
+	}
+
+	cfg.Configuration, err = readConfigurationFileIfExists()
+	if err != nil {
+		return errors.New("Error parsing configuration YAML: " + strings.TrimPrefix(err.Error(), "yaml: "))
+	}
+
+	if cfg.Currency == "" {
+		cfg.Currency = cfg.Configuration.Currency
+	}
+	if cfg.Currency == "" {
+		cfg.Currency = "USD"
+	}
+
+	return nil
+}
+
+func (c Configuration) Save() error {
+	if c.Version == "" {
+		c.Version = configurationVersion
+	}
+	return writeConfigurationFile(c)
+}
+
+func readConfigurationFileIfExists() (Configuration, error) {
+	if !fileExists(ConfigurationFilePath()) {
+		return Configuration{}, nil
+	}
+
+	data, err := ioutil.ReadFile(ConfigurationFilePath())
+	if err != nil {
+		return Configuration{}, err
+	}
+
+	var c Configuration
+
+	err = yaml.Unmarshal(data, &c)
+
+	return c, err
+}
+
+func writeConfigurationFile(c Configuration) error {
```

Comment:
For the `.state.json` we don't error, we just log an error, because that can get saved as part of the `diff` and `breakdown` commands. The `credentials.yml` and this one are only written with the `configure set` or `register` commands so I think they should show an error.

Reply:
We don't write files if the equivalent env vars are set, correct?

Reply:
I think we should always write files for the `configure set` and `register` commands. If the user has set env vars then they don't need to run these commands.